core/org.wso2.carbon.ndatasource.rdbms/src/main/java/org/wso2/carbon/ndatasource/rdbms/CorrelationLogInterceptor.java

```diff
@@ -84,7 +60,7 @@ protected void prepareCall(String s, long l) {
     @Override
     public Object createStatement(Object proxy, Method method, Object[] args, Object statement, long time) {
         try {
-            if (isEnableCorrelationLogs) {
+            if (ConfigurableCorrelationLogService.isEnable()) {
```

Comment:
Is it possible to get the registered service from the osgi level instead of using static methods?

Reply:
The static variable is used to hold the value we receive via the service. We cannot refer to the OSGi reference on the fly from here. So we have to keep it somewhere (a data holder) and then use it later.